src/Manager/Indices.php

```diff
@@ -112,24 +133,18 @@ public function generateIndexByIndexable( $indexable ) {
 		$mapping['settings']['analysis']['analyzer']['default']['language'] = $languages['analyzer'];
 		$mapping['settings']['analysis']['filter']['ewp_snowball']['language'] = $languages['snowball'];
 
-		// Set language stopwords and stemmer filters
-		$mapping['settings']['analysis']['analyzer']['default']['filter'] = array_map(
-			function( $filter ) use ( $currentIndexLanguage ) {
-				if ( 'stop' === $filter ) {
-					return 'stop_' . $currentIndexLanguage;
-				}
-				return $filter;
-			},
-			$mapping['settings']['analysis']['analyzer']['default']['filter']
-		);
-		$mapping['settings']['analysis']['analyzer']['default']['filter'][] = 'stemmer_' . $this->currentIndexLanguage;
+		// Define language stopwords
+		$stopwordFilterKey = $this->getStopwordFilterKey( $mapping['settings']['analysis']['analyzer']['default']['filter'] );
+		if ( null !== $stopwordFilterKey ) {
+			$mapping['settings']['analysis']['filter'][ $stopwordFilterKey ] = [
+				'type'        => 'stop',
```

Comment:
@decodekult, is the strong "stop" here is related to the string "stop" on line 108?
If yes, you should probably use a class constant.